README.md

```diff
@@ -193,20 +189,26 @@ msg_key
         +---> TODO
 ```
 
-`Derive` is a function which ....
+Where Derive is a function defined:
 
-:warning: needs specification :warning:
-_question - is this a good example of hkdf-expand in js: https://www.npmjs.com/package/futoin-hkdf#hkdfexpandhash-hash_len-prk-length-info-%E2%87%92-buffer ?_
+```js
+var Derive = MakeDeriver(feed_id, prev_msg_id)
 
+function MakeDeriver (feed_id, prev_msg_id) {
+  return function (key, label, length) {
+    var data = [feed_id, prev_msg_id, label]
+    return HKDF.Expand(key, encode(data), length)
+  }
+}
```

Comment:
I've written this example in js ... I think it might be good to follow your example and break out chunks of this spec into different files.

e.g. 
- README (purpose and uses of box2, broad layout of the thing)
- key derivation
  - overview with hierarchy, constants
  - Derive spec
  - SLP-encode spec
- walkthrough of encryption / decryption
- extensions